utils.py

```diff
@@ -17,3 +19,49 @@ async def fetch_plugin_json(url):
         response = await client.get(url)
         return response.json()
 
+
+def update_version_zip(name_plugin: str, version: str):
```

Comment:
This is not going to work. You should check for existance of the plugin in the list, then for the version, you can do it more comfy if you store the json and process it in memory and then you can write it in the file.

I'll rewrite this for you with some comments so you can see the process:

```python
def update_version_zip(name_plugin: str, version: str):
    cache_path = "zip_cache/versions.json"

    # Load existing versions from the JSON file
    if os.path.exists(cache_path):
        with open(cache_path, "r") as json_file:
            versions = json.load(json_file)
    else:
        versions = []

    # Search for the plugin by name_plugin
    plugin_found = False
    for plugin_info in versions:
        if plugin_info["name_plugin"] == name_plugin:
            plugin_found = True
            # Check if the version is the same
            if plugin_info["version"] != version:
                plugin_info["version"] = version
            break

    # If plugin not found, add it
    if not plugin_found:
        versions.append({"name_plugin": name_plugin, "version": version})

    # Save the updated list of versions to a variable
    updated_versions = versions

    # Write the updated list back to the JSON file
    with open(cache_path, "w") as json_file:
        json.dump(updated_versions, json_file, indent=4)
 ```